packages/contracts-bedrock/src/safe/LivenessModule2.sol

```diff
@@ -0,0 +1,226 @@
+    function clear() external {
+        // Check if the calling safe has configuration set
+        ModuleConfig storage config = safeConfigs[msg.sender];
+        if (config.fallbackOwner == address(0)) {
+            revert LivenessModule2_ModuleNotConfigured();
+        }
+
+        // Check that this module is NOT enabled on the calling Safe
+        // This prevents clearing configuration while module is still enabled
+        Safe safe = Safe(payable(msg.sender));
+        if (safe.isModuleEnabled(address(this))) {
+            revert LivenessModule2_ModuleStillEnabled();
+        }
+
+        // Erase the configuration data for this safe
+        delete safeConfigs[msg.sender];
+        // Also clear any active challenge
+        delete challengeStartTime[msg.sender];
+        emit ModuleDisabled(msg.sender);
+    }
+
+    /// @notice Returns challenge_start_time + liveness_response_period if there is a challenge for the given safe, or
+    /// 0 if not
+    /// @dev MUST never revert
+    /// @param _safe The Safe address to query
+    /// @return The challenge end timestamp, or 0 if no challenge
+    function getChallengePeriodEnd(address _safe) external view returns (uint256) {
+        uint256 startTime = challengeStartTime[_safe];
+        if (startTime == 0) {
+            return 0;
+        }
+        ModuleConfig storage config = safeConfigs[_safe];
+        return startTime + config.livenessResponsePeriod;
+    }
+
+    /// @notice Challenges an enabled safe
+    /// @dev MUST only be executable by fallback owner of the challenged safe
+    /// @dev MUST revert if there is a challenge for the safe
+    /// @dev MUST set challenge_start_time to the current block time
+    /// @dev MUST emit the ChallengeStarted event
+    /// @param _safe The Safe to challenge
+    function challenge(address _safe) external {
+        ModuleConfig storage config = safeConfigs[_safe];
+
+        if (config.fallbackOwner == address(0)) {
+            revert LivenessModule2_ModuleNotConfigured();
+        }
+
+        // Check that the module is still enabled on the target Safe
+        Safe safe = Safe(payable(_safe));
+        if (!safe.isModuleEnabled(address(this))) {
+            revert LivenessModule2_ModuleNotEnabled();
+        }
+
+        if (msg.sender != config.fallbackOwner) {
+            revert LivenessModule2_UnauthorizedCaller();
+        }
+
+        if (challengeStartTime[_safe] != 0) {
+            revert LivenessModule2_ChallengeAlreadyExists();
+        }
+
+        challengeStartTime[_safe] = block.timestamp;
+        emit ChallengeStarted(_safe, block.timestamp);
+    }
+
+    /// @notice Responds to a challenge for an enabled safe, canceling it
+    /// @dev MUST only be executable by an enabled safe
+    /// @dev MUST revert if there isn't a challenge for the calling safe
+    /// @dev MUST revert if there is a challenge for the calling safe but the response period has expired
```

Comment:
Are we sure we want this? I can imagine some scenario where for whatever reason the response comes slightly late, but the fallback owner knows that it will come, and so it 
delays on calling `changeOwnershipToFallback()`.

....

OK, with further thought, I think that we should keep this. As the operator may wish to enforce an SLA on the Safe. Leaving the comment for posterity.

Reply:
I actually agree that we might not want this. Enforcing an SLA is something we can do outside of the liveness guard. I don't see any upside to the revert, and I see downside in cases where the fallback owner is made aware that liveness can be re-established but it'll take longer than the challenge window.